### ISSUE TITLE
build.sh hangs on Mono test code - Ubuntu Desktop 14.04.3 LTS

### ISSUE THREAD

**Issue by clrjunkie** [aspnet/KestrelHttpServer#563]
Attached Build Log.
[kbuild.txt](https://github.com/aspnet/KestrelHttpServer/files/83089/kbuild.txt)

**Comment by cesarbs** [comment]
Does it happen consistently? Lots of tests are disabled on Mono precisely for this reason. We might have to disable some more.

It's really a Mono issue, since the hang happens after all tests have been executed and the xunit runner has exited.

**Comment by clrjunkie** [comment]
Sorry for the delayed response, working on multiple fronts..

>Does it happen consistently?

Yes.

>We might have to disable some more.

Consider adding an argument to build "skipmonotests"

Q: Do you have a build script that builds Kestrel directly against CoreFx build output binaries?

**Comment by cesarbs** [comment]
>Consider adding an argument to build "skipmonotests"

It's already there :grin: Set this environment variable: `KOREBUILD_TEST_SKIPMONO=1`

**Comment by davidfowl** [comment]
> Q: Do you have a build script that builds Kestrel directly against CoreFx build output binaries?

@clrjunkie No, we use packages. If you have a build that produces packages then it should be trivial to integrate using a custom feed.

**Comment by clrjunkie** [comment]
@cesarbs Thank!

@davidfowl I understand and accept but I also have further thoughts on the subject. 
I'm opening a new issue to discuss them.

**Comment by clrjunkie** [comment]
@cesarbs 

Is there a build instructions page? It might be good idea to add a list of all the environment vars controlling the build process.

**Comment by cesarbs** [comment]
@clrjunkie There's isn't one at this time, but indeed it'd be a good idea to document these things.

**Comment by clrjunkie** [comment]
@davidfowl are there any issues with running kestrel using dotnet cli? looking for the bare minimum setup to dev test outside VS

**Comment by muratg** [comment]
@cesarbs Let's continue to disable tests on Mono to unblock the CI runs.

@clrjunkie You can check out some samples for running apps with dotnet CLI here: https://github.com/davidfowl/dotnetcli-aspnet5

**Comment by clrjunkie** [comment]
@muratg

Hallelujah!

See message at the end: dotnet/cli#799

**Comment by cesarbs** [comment]
Looks like we can close this.
